### PR TITLE
[satel] Thing discovery fix

### PR DESCRIPTION
**Pull request by druciak**
This PR fixes discovery of event-log and system things in OH3. 
The issue is related to thing UID validation introduced in OH3 and both things did not pass this validation (missing bridge segment). The fix changes the way UID are created and now they are correct for both of things.

Related discussion is here: https://community.openhab.org/t/satel-binding-support-announcements-and-feature-requests/56135/245

Signed-off-by: Krzysztof Goworek <krzysztof.goworek@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/discovery/SatelDeviceDiscoveryService.java [hunk 1]
```diff
@@ -152,14 +152,10 @@ private void addDevice(DeviceType deviceType, int deviceKind, int deviceId, Stri
...
-        ThingUID thingUID;
-        if (deviceId == null) {
-            thingUID = new ThingUID(thingTypeUID, bridgeUID.getId());
-        } else {
-            thingUID = new ThingUID(thingTypeUID, bridgeUID, deviceId);
-        }
-        DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withThingType(thingTypeUID)
+        final ThingUID bridgeUID = bridgeHandler.getThing().getUID();
+        final ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID,
+                deviceId == null ? toCamelCase(thingTypeUID.getId()) : deviceId);
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/discovery/SatelDeviceDiscoveryService.java:157]: If `deviceId` is `null` the new thing will get the id based on the `thingTypeUID`. This is the same for all instances of a specific type. This means there can only be 1 per type. I don't know if that is intended, but it does look strange.

Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/discovery/SatelDeviceDiscoveryService.java:157]: Yes, that's intended. The old code was creating in such case thing UID without device ID, for the same purpose. But in OH3 this is not possible. Basically these thing types are "virtual" things, they logically group channels for specific purpose, for example accessing event log of connected alarm system. All these channels could be implemented directly in the bridge, but I decided to make it this way. Maybe this wasn't good idea, but I don't want to make a revolution now, just to fix the issue with UID.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Comment by druciak** [comment]
@cpmeister could you please have a quick look at this PR?

File: bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/SatelHandlerFactory.java [hunk 2]
```diff
@@ -80,8 +80,6 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         if (effectiveUID == null) {
             if (DEVICE_THING_TYPES_UIDS.contains(thingTypeUID)) {
                 effectiveUID = getDeviceUID(thingTypeUID, thingUID, configuration, bridgeUID);
-            } else if (VIRTUAL_THING_TYPES_UIDS.contains(thingTypeUID) && bridgeUID != null) {
-                effectiveUID = new ThingUID(thingTypeUID, bridgeUID.getId());
             }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/SatelHandlerFactory.java:83]: Why is this removed?

Comment (druciak) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/SatelHandlerFactory.java:83]: Because this part makes specific ThingUID for these thing types, which is not valid anymore. Before, for virtual thing types, thing UID was made as `satel:<virt_thing_type_id>:<bridge_id>`, because only one thing of such thing type was required to be configured. This is not possible anymore, because valid thing UID is `satel:<virt_thing_type_id>:<bridge_id>:<thing_id>`. So I removed this part and when adding new thing of virtual thing type it gets random thing ID like other types do.

Comment (cpmeister) [reply bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/SatelHandlerFactory.java:83]: Understood.

**Review by cpmeister — APPROVED** [review]
